src/lib.rs

```diff
@@ -1407,6 +1407,18 @@ impl Error {
        }
     }
 
+    /// Similar to [`io_error`] except consumes self to convert to the [`io::Error`],
+    /// if there is one.
```

Comment:
Could you rephrase this to say, "... consumes self to convert to the original [`io::Error`], if one exists."?

Reply:
Could you also add similar verbiage to the `io_error` method (that is, "return the original `io::Error`") so that they are consistent?

Could you also say that `into_io_error` may not yield the exact same `io::Error` as `io::Error::from(walkdir_err)` since the latter accounts for loop errors and may include additional contextual information?